packages/combobox/src/ComboboxOption/ComboboxOption.tsx

```diff
@@ -51,19 +68,68 @@ export const InternalComboboxOption = React.forwardRef<
         // the keyDown event does not actually fire on the option element
         e.stopPropagation();
 
-        // If user clicked on the option, or the checkbox
-        // Ignore extra click events on the checkbox wrapper
-        if (
-          !disabled &&
-          (e.target === optionRef.current ||
-            (e.target as Element).tagName === 'INPUT')
-        ) {
+        if (!disabled) {
           setSelected();
+          onClick?.(value);
         }
       },
-      [disabled, optionRef, setSelected],
+      [disabled, onClick, setSelected, value],
     );
 
+    const icon = useMemo((): JSX.Element => {
+      if (glyph) {
+        if (isComponentGlyph(glyph)) {
+          return React.cloneElement(glyph, {
+            ...glyph.props,
+            className: cx(
+              iconThemeStyles[theme],
+              {
+                [iconHighlightedStyles[theme]]: isFocused,
+                [iconDisabledStyles[theme]]: disabled,
+              },
+              glyph.props.className,
+            ),
+          });
+        }
+        console.error(
+          '`ComboboxOption` instance did not render icon because it is not a known glyph element.',
+          glyph,
+        );
+      }
+
+      return <></>;
+    }, [disabled, glyph, isFocused, theme]);
+
+    const checkbox = (
+      <Checkbox
+        aria-labelledby={optionTextId}
+        checked={isSelected}
+        tabIndex={-1}
+        disabled={disabled}
+        darkMode={darkMode}
+        className={checkBoxBaseStyles}
+      />
+    );
+
+    const checkMark = (
+      <Icon
+        glyph="Checkmark"
+        className={cx(checkMarkSizeStyle[size], checkMarkThemeStyles[theme], {
+          [checkMarkDisabledStyles[theme]]: disabled,
+        })}
+      />
+    );
+
+    const multiSelectLeftGlyph = withIcons ? icon : checkbox;
+    const multiSelectRightGlyph = withIcons && checkbox;
+
+    const singleSelectLeftGlyph = withIcons
+      ? icon
+      : isSelected
+      ? checkMark
+      : null;
+    const singleSelectRightGlyph = withIcons && isSelected && checkMark;
```

Comment:
I think extracting all this into a `getGlyphs` function would make this a bit cleaner